core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -82,47 +138,53 @@ internal class NewBookmarksDaoTest {
   }
 
   @Test
-  fun `get bookmarks from current book when zimFileReader is null`() {
-    val zimFileReader: ZimFileReader? = null
+  fun `get current zim bookmarks url`() {
+    val zimFileReader: ZimFileReader = mockk()
     val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
     every { box.query() } returns queryBuilder
+    every { zimFileReader.name } returns ""
```

Comment:
neither tests mention id and the name value you return here should be different from what the value is normally
```
    equal(BookmarkEntity_.zimId, zimFileReader?.id ?: "")
      .or()
      .equal(BookmarkEntity_.zimName, zimFileReader?.name ?: "")
```

I also wouldn't reuse the mock 

```
queryBuilder.equal(
        BookmarkEntity_.zimName,
        zimFileReader?.name ?: ""
        zimFileReader.name
      )
```

for this call, I would extract a variable

```
val expectedName = "zimName"
every {zimFileReader.name} returns expectedName
   every {
      queryBuilder.equal(
        BookmarkEntity_.zimName,
        expectedName
      )
    } returns queryBuilder
```